### PR TITLE
Atlassian

### PR DESCRIPTION
**Pull request by bradly-swart**
Functionality for handling the configuration of a custom chocolatey source authenticated or not.

### CODE REVIEW

**Comment by ferventcoder** [comment]
It seems you have backed out some changes that were just added?

**Comment by bradly-swart** [comment]
Ah I see, merge #108, I'll do a pull and merge again.

**Comment by ferventcoder** [comment]
Before you get too far down the road - this is going in a direction that only let's you add one source

**Comment by ferventcoder** [comment]
I already have a design in mind for this in #79

**Comment by bradly-swart** [comment]
Thats true, may as well handle multiple sources if you going to handle a single case.

**Comment by ferventcoder** [comment]
If you would like I could go through and comment on the bits that are not quite right - however I'm going to state right off that I've come to the conclusion this must be done as a custom type and provider because that is the only way to know the choco version on first install for sure on the client system.

**Comment by bradly-swart** [comment]
Feedback would be awesome for me personally, I like splitting it out like you have planned in #79 
We needed a super simple way of just adding a single source, so for now this suites our needs until the project has similar functionality that we can replace with.

Not keen on running on a fork for too long, so hopefully I can try and help push something out that provides this for the project.

File: manifests/init.pp [hunk 1]
```diff
@@ -13,16 +13,16 @@
...
 #   }
 #
 # @example Use a file chocolatey.0.9.9.9.nupkg for installation
 #   class {'chocolatey':
-#     chocolatey_download_url         => 'file:///c:/location/of/chocolatey.0.9.9.9.nupkg',
-#     use_7zip                        => false,
-#     choco_install_timeout_seconds   => 2700,
+#     chocolatey_download_url => 'file:///c:/location/of/chocolatey.0.9.9.9.nupkg',
+#     use_7zip                => false,
+#     choco_install_timeout   => 2700,
```
Comment (ferventcoder) [inline manifests/init.pp]: The preference is to include seconds in the name. I know it is longer but it is extremely clear that what the interval represents.

File: manifests/config.pp [hunk 2]
```diff
@@ -25,4 +27,30 @@
...
+      $_source_cmd = "${source_cmd} -u=${::source_user} -p=${::source_password}"
+    } else {
+      notify { 'source user and password not set, using unauthenticated source': }
+      $_source_cmd = $source_cmd
+    }
+
+    # Check if priority is set, default to 0
+    if $source_priority == undef{
+      notify { 'Chocolatey source priority not set, defaulting to 0': }
+      $source_priority = '0'
```
Comment (ferventcoder) [inline manifests/config.pp]: Priority can't be passed to older versions of Choco

Comment (ferventcoder) [inline manifests/config.pp]: By older, i have it listed in https://tickets.puppetlabs.com/browse/MODULES-3037

**Comment by ferventcoder** [comment]
The ability to set sources has made it in as `chocolateysource` in puppetlabs/puppetlabs-chocolatey! As previously mentioned, I don't think we would move forward on this. 

Thanks for the contribution, and hopefully the feedback was helpful for you!

**Pull request closed without merging**